src/beamlime/kafka/source.py

```diff
@@ -14,6 +14,29 @@ def close(self) -> None:
         pass
 
 
+class MultiConsumer:
+    """
+    Message source for multiple Kafka consumers.
+
+    This class allows for consuming messages from multiple Kafka consumers with
+    different configuration. In particular, we need to use different topic offsets for
+    data topics vs. config/command topics.
+    """
+
+    def __init__(self, consumers):
+        self._consumers = consumers
+
+    def consume(self, num_messages: int, timeout: float) -> list[KafkaMessage]:
+        messages = []
+        for consumer in self._consumers:
+            messages.extend(consumer.consume(num_messages, timeout))
```

Comment:
I think they shouldn't have same timeout for all different topics, unless it's very small `~0`.
If `num_message` and `timeout` are not optimized, some topics that have many messages might fall behind.